### PR TITLE
[iOS][Android][SDK] Google Sign-out

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

https://expo.canny.io/feature-requests/p/google-sign-out

Method for signing-out of `Expo.Google`

# How

https://github.com/expo/universe/compare/@janic/google-signout

### CODE REVIEW

File: android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/GoogleModule.java [hunk 1]
```diff
@@ -117,6 +117,13 @@ public void run() {
     });
   }
 
+  @ReactMethod
+  public void signOut() {
```
Comment (ide) [inline android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/GoogleModule.java]: Make this an async function -- even though there is no value returned, returning from a function (i.e. resolving a promise in the async case) also conveys that the function finished running.

**Review by janicduplessis — APPROVED** [review]

**Review by esamelson — COMMENTED** [review]
@EvanBacon thanks for taking this on! I am curious if you were able to successfully test this. I ask because I tried running the code from @janicduplessis 's branch before and was able to call the `signOut` method, but it didn't seem to actually do anything (e.g. sign in still behaved like it was already signed in, and the auth token I had still fetched my google user data).

**Comment by janicduplessis** [comment]
@esamelson Have you tested as a standalone app? It is only implemented for the native sdk based flow. I remember looking into implementing it for the web based flow too but seemed more complex to do.

**Comment by EvanBacon** [comment]
@esamelson this isn't perfect, I'll prolly just close this PR and wait until `AppAuth` is done. Going to try and open `GoogleSignIn` today.

**Comment by EvanBacon** [comment]
Closing in favor of #2565 #2543

**Pull request closed without merging**